### ISSUE TITLE
Moving from 9.3.8 to 9.4.{1 or 3} PostgreSQL data sources are losing their database names

### ISSUE THREAD

**Issue by In-At-Pure** [grafana/grafana#64273]
**What happened**:
Moving from 9.3.8 to 9.4.1 (or 9.4.3) none of my Postgresql data sources work because their database name fields are empty (i.e. were not loaded correctly from file). I'm provisioning the data source descriptions via YAML files (all of which have the correct database names specified).

**What you expected to happen**:
I expect the database name in the YAML file to work as it did previously.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
I note that if I create a new posgresql data source it works as expected BUT if I dump that data source via api/datasources the database field is emitted with its value as the empty string.

**Environment**:
- Grafana version: 9.4.1
- Data source type & version:
- OS Grafana is installed on: docker container
- User OS & Browser:
- Grafana plugins:
- Others:

**Comment by tonypowa** [comment]
:wave: Thanks for creating this issue, @In-At-Pure 

could you share your data source config file? make sure there's not sensitive data in it

also, what  is the `name` and `uid` for PostgresSQL when you call the api endpoint `api/datasources`  ? could you share the api response?

thank you 👍

**Comment by Orson-Cart** [comment]
Below is additional data as requested, these are not my original datasource entries (obviously) but I did confirm that they manifest the problem. In preparing this data I see something interesting. I notice that in the JSON dumped from the api, there are 2 "database" entries, one at the top level (which is empty) and one in the jsonData section (which has the expected value). I see that in my original YAML file (which came from an 8x Grafana release), there is a single database entry (which has the correct value in the YAML but which is not loaded). Furthermore I find that if I add a suitable "database" entry to the jsonData section of the YAML file then that value is loaded correctly as the database name. So I'd guess that the database field moved between releases and there is now a compatibility/consistency issue.

- JSON from the api dump for a psql database with a database name of foo-bar

```
  {
    "id": 16,
    "uid": "BXmZEW-Vk",
    "orgId": 1,
    "name": "DEMO",
    "type": "postgres",
    "typeName": "PostgreSQL",
    "typeLogoUrl": "public/app/plugins/datasource/postgres/img/postgresql_logo.svg",
    "access": "proxy",
    "url": "foo.bar.com",
    "user": "readonly",
    "database": "",
    "basicAuth": false,
    "isDefault": false,
    "jsonData": {
      "database": "foo-bar",
      "sslmode": "disable"
    },
    "readOnly": false
  }

```
- YAML for a psql DB, when this is loaded the database name field (in Grafana) is empty.

```
---
apiVersion: 1
datasources:
  - access: proxy
    basicAuth: false
    database: foo_bar
    id: 3
    isDefault: false
    jsonData:
      postgresVersion: 905
      sslmode: disable
      tlsAuth: false
      tlsAuthWithCACert: false
      tlsConfigurationMethod: file-path
      tlsSkipVerify: true
    name: DEMO
    orgId: 1
    readOnly: true
    type: postgres
    typeLogoUrl: public/app/plugins/datasource/postgres/img/postgresql_logo.svg
    typeName: PostgreSQL
    uid: DeadBeef
    url: foo.bar.com
    user: readonly
    secureJsonData:
      password: 'NothingToSeeHereFolks'

```

**Comment by marcsello** [comment]
I have run into a similar issue with Grafana 9.4.3 with InfluxDB as datasource. The database name wasn't imported from the provisioned config resulting in a failure to use said datasource.

The datasource name was provided in the provisioned config:
`/etc/grafana/provisioning/datasources/datasources.yaml`
```yaml
apiVersion: 1
datasources:
- database: my-database
  isDefault: true
  name: InfluxDB
  type: influxdb
  url: http://my-influxdb:8086
- jsonData:
    httpHeaderName1: my-header
  name: Loki
  secureJsonData:
    httpHeaderValue1: "my-val"
  type: loki
  url: http://my-loki:3100
```
Other info and datasource are imported fine, only the database name from the InfluxDB datasource is not imported.


Reverting to Grafana version 9.3.8 fixes the issue.

Note: We are using the official Grafana Helm chart for deploying Grafana in k8s, but It seems to be irrelevant as the config is mounted properly as it should, and haven't changed between chart versions.

**Comment by In-At-Pure** [comment]
Below is the additional data as requested. This is not my original data (obviously) but I did confirm that this manifests the problem. While preparing this data I noticed the following; The JSON dumped from the api has 2 "database" entries, one at the top level (which has an empty string value) and one in the "jsonData" section (which has the expected value). I also note that in the YAML file (which was derived from an 8.x dump) there is a single "database" entry (which is correct but which does not get loaded as expected). Furthermore I find that if I add an appropriate "database" entry into the "jsonData" section of the YAML file then the correct value is loaded. So I'd guess that the database data moved between releases creating a consistency/compatibility issue.

- JSON from the api dump of a psql DB with a database name of "foo-bar":

```
   {
    "id": 16,
    "uid": "BXmZEW-Vk",
    "orgId": 1,
    "name": "DEMO",
    "type": "postgres",
    "typeName": "PostgreSQL",
    "typeLogoUrl": "public/app/plugins/datasource/postgres/img/postgresql_logo.svg",
    "access": "proxy",
    "url": "foo.bar.com",
    "user": "readonly",
    "database": "",
    "basicAuth": false,
    "isDefault": false,
    "jsonData": {
      "database": "foo-bar",
      "sslmode": "disable"
    },
    "readOnly": false
  }

```
- YAML description of a psql datasource with a database name of "foo_bar":

```
---
apiVersion: 1
datasources:
  - access: proxy
    basicAuth: false
    database: foo_bar
    id: 3
    isDefault: false
    jsonData:
      postgresVersion: 905
      sslmode: disable
      tlsAuth: false
      tlsAuthWithCACert: false
      tlsConfigurationMethod: file-path
      tlsSkipVerify: true
    name: DEMO
    orgId: 1
    readOnly: true
    type: postgres
    typeLogoUrl: public/app/plugins/datasource/postgres/img/postgresql_logo.svg
    typeName: PostgreSQL
    uid: DeadBeef
    url: foo.bar.com
    user: readonly
    secureJsonData:
      password: 'NothingToSeeHereFolks'

```

**Comment by tonypowa** [comment]
thanks for the input @marcsello  @In-At-Pure  🤟 

I'm forwarding the issue to the engineering team that supports PostgreSQL as data source

**Comment by zoltanbedi** [comment]
This is fixed in 9.4.8. We moved the database field under jsonData https://github.com/grafana/grafana/pull/58649 but it got a bug in it with provisioning. https://github.com/grafana/grafana/pull/58649
